Story1

Sen-i opened the pull request.

Sen-i committed:
setup db and configfile

Sen-i committed:
testing brances

Sen-i committed:
display data from DB

Sen-i committed:
changes as suggested to formatting

Sen-i committed:
Cleanup

Sen-i committed:
Merge branch 'story1' into dbsetup
A

Sen-i committed:
changed db

Sen-i committed:
Merge pull request #2 from Sen-i/dbsetup

Dbsetup

Sen-i committed:
removed repo within reps

Sen-i committed:
display o/p from db

Sen-i committed:
Merge pull request #3 from Sen-i/appDesign

display o/p from db

Sen-i committed:
added css styling

Sen-i committed:
update DB≈

Sen-i committed:
Merge pull request #4 from Sen-i/webDesign

added css styling

Sen-i committed:
Split program into functions
Unit test added

Sen-i committed:
Added line breaks at the end of files

Sen-i committed:
style changes

Sen-i committed:
Beautifying code and fixing space btw functions

Sen-i committed:
Merge pull request #5 from Sen-i/unitTests

Unit tests

mporam commented on config.php:
```diff
@@ -0,0 +1,27 @@
+<?php
+
+//$db = new PDO('mysql:host=DB;dbname=shapeDB','root','password');
```
please remove these lines

mporam commented on config.php:
```diff
@@ -0,0 +1,27 @@
+/**
+ * @return PDO A database instance
+ */
+
```
comment should be directly next to the function

mporam commented on config.php:
```diff
@@ -0,0 +1,27 @@
+ */
+
+function getFromDatabase(PDO $database): array {
+    $query = $database->prepare("SELECT * FROM `shapes`");
```
`*` is bad, its really slow. please list the field names

mporam commented on getFunctions.php:
```diff
@@ -0,0 +1,44 @@
+ */
+
+function getNameImg(array $shape): string {
+    return "<img alt=\"" . $shape["ShapeName"] . "\"src=\"" . $shape["imgurl"] ."\" />";
```
when you need to use quotes inside a string, wrap the string in the opposite kind of quotes so you dont have to escape it. so here, use single quotes to contain the string. keeps it cleaner

mporam commented on getFunctions.php:
```diff
@@ -0,0 +1,44 @@
+ * @return string The name of the shape
+ */
+
+function getShapeName(array $shape): string {
```
all of these get functions (except the image one) dont actually do anything, you can just remove them

mporam commented on index.php:
```diff
@@ -0,0 +1,46 @@
+<?php
+
+require_once "config.php";
+require_once "./getFunctions.php";
```
you dont need the `./` - keep it consistent so either use it on both or neither

mporam commented on config.php:
```diff
@@ -0,0 +1,27 @@
+ * @return array The shapes in the database.
+ */
+
+function getFromDatabase(PDO $database): array {
```
this isnt a semantic function name. get what from database?? why not just `getShapes`. doesnt matter where they come from - encapsulation

mporam commented on test/dataCollectionAppTest.php:
```diff
@@ -0,0 +1,84 @@
+
+class dataCollectionAppTest extends TestCase {
+
+    public function testgetNameImg() {
```
this is a success test so name is something like `testgetNameImg_success`

mporam commented on test/dataCollectionAppTest.php:
```diff
@@ -0,0 +1,84 @@
+        $this->assertSame($expected, $actual);
+    }
+
+    public function testErrorWhenString() {
```
you need to specify the function you are testing in the name, try `testgetNameImg_error_string`

mporam commented on test/dataCollectionAppTest.php:
```diff
@@ -0,0 +1,84 @@
+        getNameImg("For the Error!");
+    }
+
+    public function testErrorWhenEmpty() {
```
make consistent with above comments

mporam commented on test/dataCollectionAppTest.php:
```diff
@@ -0,0 +1,84 @@
+        $this->assertSame($expected, $actual);
+    }
+
+    public function testErrorWhenString2() {
```
if you name your tests as mentioned above, you wont need to number your tests, which makes them much easier to understand. although you dont actually need this test as you dont need the function

Sen-i committed:
removed redundant functions
renamed DB connection function to be more semantic
changed file name in require to be consistent
changed SQL query from * to field name for performance
changed quotes within strings to avoid escaping them
removed outdated comments

Sen-i committed:
Merge pull request #7 from Sen-i/unitTests

removed redundant functions

Sen-i committed:
css tweaks for visability
